web/packages/design/src/Modal/Modal.tsx

```diff
@@ -16,15 +16,84 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
-import styled from 'styled-components';
-import PropTypes from 'prop-types';
+import React, { createRef, cloneElement } from 'react';
+import styled, { StyleFunction } from 'styled-components';
+import { createPortal } from 'react-dom';
 
-import { ownerDocument } from './../utils';
-import Portal from './Portal';
-import RootRef from './RootRef';
+type Props = {
+  /**
+   * If `true`, the modal is open.
+   */
+  open: boolean;
+  className?: string;
+  /**
+   * Styles passed to the modal, the parent of the children.
+   */
+  // TODO(ravicious): The type for modalCss might need some work after we migrate the components
+  // that use <Modal> to TypeScript.
+  modalCss?: StyleFunction<any>;
+  children?: React.ReactElement;
+  /**
+   * Properties applied to the [`Backdrop`](/api/backdrop/) element.
+   *
+   * invisible: Boolean - allows backdrop to keep bg color of parent eg: popup menu
+   */
+  BackdropProps?: object;
+  /**
+   * If `true`, the modal will not automatically shift focus to itself when it opens, and
+   * replace it to the last focused element when it closes.
+   * This also works correctly with any modal children that have the `disableAutoFocus` prop.
+   *
+   * Generally this should never be set to `true` as it makes the modal less
+   * accessible to assistive technologies, like screen readers.
+   */
+  disableAutoFocus?: boolean;
+  /**
+   * If `true`, clicking the backdrop will not fire any callback.
+   */
+  disableBackdropClick?: boolean;
+  /**
+   * If `true`, the modal will not prevent focus from leaving the modal while open.
+   *
+   * Generally this should never be set to `true` as it makes the modal less
+   * accessible to assistive technologies, like screen readers.
+   */
+  disableEnforceFocus?: boolean;
+  /**
+   * If `true`, hitting escape will not fire any callback.
+   */
+  disableEscapeKeyDown?: boolean;
+  /**
+   * If `true`, the modal will not restore focus to previously focused element once
+   * modal is hidden.
+   */
+  disableRestoreFocus?: boolean;
+  /**
+   * If `true`, the backdrop is not rendered.
+   */
+  hideBackdrop?: boolean;
+  /**
+   * Callback fired when the backdrop is clicked.
+   */
+  onBackdropClick?: (event: React.MouseEvent) => void;
+  /**
+   * Callback fired when the component requests to be closed.
+   * The `reason` parameter can optionally be used to control the response to `onClose`.
+   */
+  onClose?: (
+    event: React.UIEvent,
```

Comment:
Depending on whether it was due to a click or keypress, the handler gets either a React event, or a native event (look how the event handlers are attached).